### PR TITLE
1270 add UI images

### PR DESCRIPTION
**Pull request by DiogoMendonc-a**
# Description
Adds a Image asset that loads images from files using stb_image.
Adds a UIImage component that uses an Image asset to store a Texture and draw it to screen.
Adds a UINativeAspectRatio that changes an element's size to fit the native aspect ratio.

## Checklist

- [x] Self-review changes.
- [x] Evaluate impact on the documentation.
- [ ] Ensure test coverage.
- [x] Write new samples.
- [x] Add entry to the changelog's unreleased section.

### CODE REVIEW

File: engine/src/ui/image/plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+using namespace cubos::engine;
+
+namespace
+{
+    struct PerElement
+    {
+        glm::vec2 xRange;
+        glm::vec2 yRange;
+        glm::vec4 color;
+        int depth;
```
Comment (RiscadoA) [inline engine/src/ui/image/plugin.cpp:24]:
```suggestion
        int depth;
        float padding[3];
```
Necessary to work with WebGL (struct size must be a multiple of 16 bytes)

File: engine/src/ui/image/plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            {
+                if (image.texture == nullptr)
+                {
+                    auto asset = assets.read<Image>(image.asset);
+                    ratio.ratio = (float)asset->width / (float)asset->height;
+                }
+            }
+        });
+
+    cubos.system("draw image")
```
Comment (RiscadoA) [inline engine/src/ui/image/plugin.cpp:99]:
```suggestion
    cubos.system("draw UI images")
```

File: engine/src/ui/image/plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        .tagged(assetsTag)
+        .after(windowInitTag)
+        .call([](Commands cmds, const Window& window, const Assets& assets) {
+            auto& rd = window->renderDevice();
+            auto vs = assets.read(VertexShader)->shaderStage();
+            auto ps = assets.read(PixelShader)->shaderStage();
+            cmds.emplaceResource<State>(rd, rd.createShaderPipeline(vs, ps));
+        });
+
+    cubos.system("set fixed aspect ration")
```
Comment (RiscadoA) [inline engine/src/ui/image/plugin.cpp:84]:
```suggestion
    cubos.system("set UI Image native aspect ratio")
```

File: engine/src/ui/image/plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    cubos.component<UIImage>();
+
+    cubos.depends(uiCanvasPlugin);
+    cubos.depends(windowPlugin);
+    cubos.depends(shaderPlugin);
+    cubos.depends(assetsPlugin);
+
+    cubos.uninitResource<State>();
+
+    cubos.startupSystem("setup color rect")
```
Comment (RiscadoA) [inline engine/src/ui/image/plugin.cpp:74]:
```suggestion
    cubos.startupSystem("setup UI Image")
```

**Review by RiscadoA — APPROVED** [review]
Just some nitpicks, otherwise LGTM! :tada: Thank you for working on this!

**Review by Scarface1809 — APPROVED** [review]

**Review by teres4 — APPROVED** [review]

**Pull request merged**